<html><head>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<title>ATOL Phyloinformatics Issues</title></head><body bgcolor="#FFFFFF"><h2><a href="index.html">ATOL</a> Phyloinformatics issues</h2><h3><a href="mailto:r.page@bio.gla.ac.uk?Subject=ATOL">Rod Page</a></h3><p>&nbsp;</p><p>Version 0.1 13 Nov 2002</p><p>These notes are cobbled together from notes made at the Kew meeting. They need   not in any way represent the views of the ATOL group, and are intended as a   starting point for discussion.</p><h3>Deliverables</h3><ol>  <li>A database of data matrices (sequence, gene order, morphology, etc.) and     trees, together with sophisticated, flexible Web interface supporting both     predefined and user-specified queries. The database would interface to all     relevant databases, such as the Catalogue of Life (<a href="http://www.sp2000.org/">Species2000</a>,     <a href="http://www.itis.usda.gov/">ITIS</a>), GenBank, <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed">Pubmed</a>,     etc.</li>  <li>Tools to create a summary tree of life. These will include supertree methods,     and online methods. The tree will be automatically updated as new data is     added.</li>  <li>Provide access to tools to build subtrees (e.g., analyses of individual     data sets, or multiple datasets such as might result from a query to the database).     Either provide computational resources for this, or provide tools and interface     to perform these computations over the GRID.</li>  <li>Release working drafts of the ToL, each with a release number so the particular     tree used can be identified in publications (e.g., "we computed the correlation     between body mass and parasite load using independent contrasts on the mammal     phylogeny contained in release 1.02 of the tree of life").</li>  <li>A cohort of trained systematists.</li></ol><h3>Some basic principles</h3><ol><li>All data considered by at least some systematists to be phylogenetically relevant should be included, i.e., sequences, morphology, behaviour, gene order/composition, DNA hybridisation, immunological methods, allozymes, RAPDs, etc. There is no scope for data or methodological chauvinism.</li>  <li>Data submission must be as painless as possible, and use (at least as a starting     point) currently accepted data standards (e.g., NEXUS, PHYLIP, Hennig86, <a href="http://biodiversity.uno.edu/delta/">DELTA</a>).</li>  <li>Data and tree output must at a minimum support standards used by the systematics     community (for example,<a href="http://www.ebi.ac.uk/embl/Submission/align_top.html">     EMBL-Align's</a> default format is unique to that database and thereby it     limits the utility of that database). If and when XML becomes used in the     systematics community it will be supported (at present there is a lot of conversation     and not much action).</li><li>The data must be acessible in ways other than  through a web interface, e.g. the database must provide XML-based web services, and the entire database must be able to be downloaded.</li><li>The database is more than an archive. It must enable people to do useful "meta" research.</li></ol><h3>Housing the ToL project</h3><p>One approach is to establish a "phyloinformatics centre". It would have a Director (a scientist) with support scientific, administration and technical staff. The Centre would house the database, and the programmers and technical support. The centre would have a small scientific staff, but would have a vistor programme where members of the ToL effort could visit for short to long periods to undertake specific research (along the lines of the DIMACS Center at Rutgers, or the Newton Centre at Cambridge). The Centre would convene regular workshops for ToL members and others (e.g., members of the US ToL effort) to evaluate progresses, and identify new topics for research. The Centre would also run training workshops in phylogenetics for graduate and postdoctoral researchers, and courses on using the ToL database and its applications.</p><p>Alternatively, perhaps we don't actually need a centre (and all the issues that raises). Perhaps if the database is linked to an established bioinformatics centre, then a separate phyloinformatics centre is unneccessary. Alternatively, the "centre" could be housed at an existing systematics institution and not actually be physically associated with the database.</p><p>In looking the Centre there are at least two major considerations. Firstly, the long term future of the database (i.e., beyond the end of FP6 funding) needs to be ensured. Secondly, it needs to have access to a community of people experienced in establishing and maintaining large biological databases. One obvious candidate is the EBI at Hinxton, Cambridge. We would need some formal committment from them to house and support the database. (Initial discuussions have been encouraging)</p><h2>Other issues (in no particular order)</h2><h3>Intellectual Property</h3><p>All data and results should be freely available without restriction of access. Need to think about whether commercial exploitation that used the data but did not make the results freely available would be allowed. In my opinion this would be highly undesirable.</p><p>All software tools developed for the project should be under GPL or an equivalent license, and if possible only Open Source tools would be used.</p><h3>Sponsors</h3>The computational aspects might draw sponsorship, say from IBM or another hardware/software vendor. Might be attractive (especially if they provided hardware support), providing no limitations result from this (i.e., not committed to proprietary software). IBM at least, seem clued up about Open Source issues.<h3>Computer science</h3><p>Although the ToL group includes (at least) 2 bona fide computer scientists (Paul & Vincent) it might be useful to have computer scientists/bioinformaticians with expertise in database design and implementation. This would strengthen our proposal. Whereas we have demonstrable expertise in phylogenetic algorithms and software development, database expertise is not a strength.</p><h3>Curation</h3><p>All aspects of the ToL database will need curation. The database itself will need to be backed up, the software (programs, scripts, etc.) will need to be maintained, debugged and improved as the project evolves. The hardware will need to be maintained.</p><p>The is also the issue of data entry, annotation, and quality control. Some of this can be automated, but much will need human intervention, certainly in the early stages of retrospective data capture.</p><h3>Data capture and entry</h3><p>Initially a database could be populated using currently available data, such   as GenBank <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=Popset">Popsets</a>,   EMBL-Align, <a href="http://www.treebase.org/">TreeBASE</a>, journal web sites.   In some cases the data can be obtained automatically and could be parsed into   the database (e.g., PopSets). If EBI are on board, then we could probably get   data straight from EMBL-Align.</p><p>There is a lot of data that is not available in electronic form. There should be an effort to extract this from the literature. OCR tools can be used to extract alignments, and TreeThief can be used to get Newick style tree descriptions from pictures of trees. There is scope for automating this.</p><h3>Subprojects</h3>We could identify a set of specific tasks/subprojects that will require research. These could be part of the activities in the first 18 months. These might include:<ol><li>data capture (i.e., from existing databases and publications)</li><li>data submission tools</li><li>supertree methods</li><li>"online" algorithms</li><li>querying tree structures</li><li>reconciling competing taxonomies for ToL navigation</li><li>visualising and navigating big trees</li></ol><h3>Relationship to other data bases</h3><p>The relationship to other taxonomic and genomic databases should be explored. Some parts of the ToL database will rely heavily on external databases. This is good in that it reduces the ToL workload, but if we rely on some services (such as a taxon name server like ITIS) it would be good to ensure any communication protocol we rely on will either persist, or we would be notified if it changed).</p><h3>Outreach</h3><p>Kare Bremer made the point that the ToL needs to be seen as the authorative   repository of phylogenetic information, so that, for example, journals would   agree to enforcing data submission. In other words, we need the same status   as GenBank. Perhaps need to negotiate with journal editors. Maybe try and get   as many European-based editors on board. </p><p align="center">[&lt;- <a href="index.html">Back to ATOL home page</a>]</p></body></html>